applications/app/controllers/CampaignsController.scala

```diff
@@ -0,0 +1,47 @@
+package controllers
+
+import common._
+import conf.Configuration
+import model._
+import play.api.libs.json._
+import play.api.libs.ws.{WSClient, WSResponse}
+import play.api.mvc._
+
+import scala.concurrent._
+
+class CampaignsController(
+  val controllerComponents: ControllerComponents,
+  val wsClient: WSClient,
+)(implicit context: ApplicationContext)
+  extends BaseController with ImplicitControllerExecutionContext with Logging {
+
+  val root = Configuration.formstack.url + "/form/"
+  val endpoint = "/submission.json"
+  val token = Configuration.formstack.editorial.oauthToken
+
+  def formSubmit() = Action.async { implicit request: Request[AnyContent] =>
+
+    val pageUrl: String = request.headers("referer")
```

Comment:
It shouldn't be, but at worst could redirect to guardian homepage

Reply:
This throws a type error: found String, required Boolean - I don't understand why?
```
    val pageUrl: String = if(request.headers("referer"))request.headers("referer") else "https://www.theguardian.com/uk"
```

Reply:
Because `request.headers("referer")` is a String. Although you can use it in an `if` expression in JavaScript, you can't do the same in Scala or strongly-typed languages. You probably want to use `if (request.hasHeader("referer")) ...`